### DISCUSSION TITLE
ImageMagick

### DISCUSSION THREAD

**Discussion by mauricirolo** [ImageMagick/ImageMagick#6651] [Help]
### ImageMagick version

7.1.1-16

### Operating system, version and so on

Windows

### Description

Hello!

I'm currently working on a project involving ImageMagick, where I'm attempting to give all images a grey background. Unfortunately, I've encountered an issue with the following codes:

- convert input.jpg -background gray -resize 2000x2000 -gravity center -extent 2000x2000 output.jpg
- convert input.jpg -resize 2000x2000 -background "rgb(160,160,255)-gravity center -extent 2000x2000 output.jpg
- convert input.jpg -resize 2000x2000 -background gray -fill gray -gravity center -extent 2000x2000 output.jpg

It appears that the image background isn't changing as expected. Could someone kindly assist me in resolving this issue? Your help would be greatly appreciated!

**Comment by snibgo** [comment]
I do not recommend using a `convert` command for any purpose unless you really have to, eg a legacy system that cannot be upgraded. `convert` is the old ImageMagick v6. For some years now, v7 has been available. The v7 syntax is (slightly) more logical, and more powerful, and has better error checking. Use `magick` instead of `convert`. And don't use `magick convert` as that uses v7 software with v6 syntax, for backwards compatibility.

None of your commands will change pixel colours. Perhaps the `-opaque` operation would help. See https://imagemagick.org/script/command-line-options.php#opaque

Your middle command is bad: there is a single quotation mark.

**Comment by fmw42** [comment]
-background is a setting not an operator. It sets the background color when an appropriate operator is supplied such as -extent.  That adds the background color only to extended pixels.
